### ISSUE TITLE
Exhaustiveness check fails for object patterns with typos

### ISSUE THREAD

**Issue by bdrobinson** [gvergnaud/ts-pattern#44]
**Describe the bug**
When pattern matching against multiple attributes in an object, `ts-pattern` will throw a runtime error that the exhaustive pattern matching was not able to spot at build-time if one of the attributes is misspelled.

**Code Sandbox with a minimal reproduction case**
https://codesandbox.io/s/elated-wilbur-h0mz2?file=/src/index.ts
(copy+pasted too):
```ts
import { match } from "ts-pattern";

type Person = {
  sex: "Male" | "Female";
  age: "Adult" | "Child";
};

function summary(person: Person): string {
  return (
    match(person)
      // Typo – "agf" should be "age"
      .with({ sex: "Female", agf: "Adult" }, () => "Woman")
      .with({ sex: "Female", age: "Child" }, () => "Girl")
      .with({ sex: "Male", age: "Adult" }, () => "Man")
      .with({ sex: "Male", age: "Child" }, () => "Boy")
      // there is no TS error
      .exhaustive()
  );
}

// yet this throws an error at runtime :(
console.log(summary({ sex: "Female", age: "Adult" }));
```

**Versions**
- TypeScript version: 4.3.5
- ts-pattern version: 3.2.4
- environment: browser + version / node version / deno version

Thanks – this library is great!

**Comment by gvergnaud** [comment]
Thanks, I'll take a look

**Comment by gvergnaud** [comment]
This has been fixed in [v3.2.5](https://github.com/gvergnaud/ts-pattern/releases/tag/v3.2.5)! 

Code sandbox: https://codesandbox.io/s/nifty-moore-pwxx7?file=/src/index.ts
